### PR TITLE
Refactored code to include timestamp as part of transaction

### LINKED ISSUE #100: Refactor transactions to include timestamp
**Description:**

Remove generating timestamp when creating block and instead generate timestamp in client and send it with the transaction. Also update hashing in frontend and backend to use timestamp as an input alongside the other transaction data.

- new Transaction object: 

```json
{

"to":"toAddress",
"from":"fromAddress",
"amount": 5,
"timestamp":Time


}
```

**Acceptance Criteria:**

- [x] Update transaction models
- [x] Remove shared transaction model in backend
- [x] Update hashing
- [x] Update relevant tests

### PR DESCRIPTION
**Pull request by ZackHolmberg**
closes #100

### CODE REVIEW

File: src/backend/services/mining-api/src/routes.py [hunk 1]
```diff
@@ -70,8 +60,10 @@ def add_data_to_queue():
 
 def valid_proof(hash_, nonce) -> bool:
     valid = (hash_.startswith('0' * difficulty))
+    print("TRANSACTION: ", ongoing_transaction)
```
Comment (ZackHolmberg) [inline src/backend/services/mining-api/src/routes.py]: Leftover print

File: src/frontend/src/App.vue [hunk 2]
```diff
@@ -73,6 +74,7 @@ export default class App extends Vue {
             "minerId": this.walletId
           }
 
+          console.log("Sending: ",toSend, " corresponding to transaction:", transaction)
```
Comment (ZackHolmberg) [inline src/frontend/src/App.vue]: Leftover log

**Review by madisonfines — APPROVED** [review]
office2

**Review by abhisachdev17 — APPROVED** [review]
Looks good to me, while you are at it can you make sure ids are unique for each transaction? you just need to call uuid function to get unique ids.

**Comment by abhisachdev17** [comment]
> Looks good to me, while you are at it can you make sure ids are unique for each transaction? you just need to call uuid function to get unique ids.

Oh lol nevermind, I just saw realized ids will always be unique! My bad, its good to merge

**Pull request merged**